### PR TITLE
feat: Add owner field to Entity and rename labels to tags

### PR DESCRIPTION
**Pull request by felixwang9817**
Signed-off-by: Felix Wang <wangfelix98@gmail.com>

**What this PR does / why we need it**: This PR adds the `owner` field to `Entity`. It also renames `labels` to `tags`, and cleans up some unused methods.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Owner field added to Entity, and labels renamed to tags
```

### CODE REVIEW

File: sdk/python/feast/entity.py [hunk 1]
```diff
@@ -97,88 +95,68 @@ def __str__(self):
...
-        Gets the description of this entity.
-        """
-        return self._description
+    def value_type(self) -> ValueType:
+        return self._value_type
 
-    @description.setter
-    def description(self, description):
-        """
-        Sets the description of this entity.
```
Comment (woop) [inline sdk/python/feast/entity.py:122]: Is there a reason we are removing all comments?

File: sdk/python/feast/entity.py [hunk 2]
```diff
@@ -97,88 +95,68 @@ def __str__(self):
...
-    def labels(self) -> Dict[str, str]:
-        """
-        Gets the labels of this entity.
-        """
-        return self._labels
+    def tags(self) -> Dict[str, str]:
+        return self._tags
 
-    @labels.setter
-    def labels(self, labels: Dict[str, str]):
```
Comment (woop) [inline sdk/python/feast/entity.py:162]: Isn't this a breaking change? I think we should keep backwards compat for at least one minor version.

File: sdk/python/feast/entity.py (hunk 1 above)
Comment (felixwang9817) [reply sdk/python/feast/entity.py:122]: as discussed, we will remove all properties

File: sdk/python/feast/entity.py (hunk 2 above)
Comment (felixwang9817) [reply sdk/python/feast/entity.py:162]: yeah good catch - just updated this

**Review by woop — APPROVED** [review]

**Pull request merged**